ts/database.ts

```diff
@@ -0,0 +1,98 @@
+//this module uses UUID v8.1.0 library (CDN). The uuidv4 function comes from there
+import { MILISECOND_HOUR, getDayStart } from "./dateManipulation.js";
+import { TIMESLOT_DURATION } from "./weekView.js";
+import { storageState } from "./state.js";
+
+declare global {
+	interface Window {
+		uuidv4: any;
+	}
+}
+
+export const filterEventsByTimestamp = (timestamp: number): FormData[] => {
+	const events = getEvents();
+	return events.filter((event: FormData) => {
+		return getEventCellTimestamp(event) === timestamp;
+	});
+};
+
+export const findEventById = (eventId: string): FormData | null => {
+	const events = getEvents();
+	const result = events.find((event: FormData) => {
+		return event.id === eventId;
+	});
+	if (!result) {
+		return null;
+	}
+	return result;
+};
+
+export class FormData {
+	id: string;
+	title: string;
+	description: string;
+	startTime: number;
+	endTime: number;
+	constructor(data: any) {
+		this.id = (data.id as string) || window.uuidv4();
+		this.title = data.title;
+		this.description = data.description;
+		this.startTime = data.startTime;
+		this.endTime = data.endTime;
+	}
+}
+
+export const getEventDuration = (event: FormData): number => {
+	return Math.ceil((event.endTime - event.startTime) / (MILISECOND_HOUR / 4));
+};
+export const getEventTimeslot = (event: FormData): number => {
+	const startDate = new Date(event.startTime);
+	return Math.floor(startDate.getMinutes() / TIMESLOT_DURATION);
+};
+export const getEventCellTimestamp = (event: FormData): number => {
+	return new Date(event.startTime).setMinutes(0).valueOf();
+};
+export const getEventStartDate = (event: FormData): number => {
+	return getDayStart(event.startTime);
+};
+export const getEventEndDate = (event: FormData): number => {
+	return getDayStart(event.endTime);
+};
+
+const getEvents = (): FormData[] => {
+	const eventsString = localStorage.getItem("events");
+	if (eventsString) {
+		const parsedEvents: object[] = JSON.parse(eventsString);
+		const events: FormData[] = parsedEvents.map((event) => new FormData(event));
+		return events;
+	}
+	return [];
+};
+
+export const saveFormData = (formData: FormData): FormData[] => {
+	const events: FormData[] = getEvents();
+	events.push(formData);
+	setStorage("events", events);
```

Comment:
It's generally better to manage your state without mutating it.
Instead of pushing to the array and making a mutation on the existing object, you could save an entirely new array as a result:
`setStorage("events",[...events, formData]);`
Read about immutability and why it's important if you haven't heard about this topic.